sklearn/utils/estimator_checks.py

```diff
@@ -1191,6 +1192,58 @@ def check_estimators_pickle(name, estimator_orig):
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
 
+def check_pairwise_estimator_tag(name, estimator_orig):
+    attributes_to_check = [('metric', 'precomputed'),
+                           ('affinity', 'precomputed'),
+                           ('kernel', 'precomputed')]
+
+    # Check if _pairwise attribute is present - will be used later
+    has_pairwise_tag = (False
+                        if getattr(estimator_orig, '_pairwise', None) is None
+                        else True)
+
+    # Using iris as sample data
+    iris = load_iris()
+    X, y_ = iris.data, iris.target
+    distance_matrix = pairwise_distances(X)
+
+    for attribute, attribute_value in attributes_to_check:
+        # Check to see if attribute value is supported by estimator
+        if attribute not in estimator_orig.get_params():
+            continue
+        try:
+            # Construct new object of estimator with desired attribute value
+            modified_estimator = clone(estimator_orig).set_params(
+                                                **{attribute: attribute_value})
+            # Not all estimators validate parameters, so check fit()
```

Comment:
Most estimators do not. Rather comment that "Estimators may validate parameters in fit if not in set_params"

Reply:
Resolved